src/App.tsx

```diff
@@ -1,61 +1,134 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const getTodoId = (todos: Todo[]) => {
+  const id = Math.max(...todos.map(todo => todo.id));
+
+  return id + 1;
+};
+
+const getUser = (userId: number) => (
+  usersFromServer.find(user => user.id === userId) || null
+);
+
+const todosAndUsers: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosAndUsers);
+  const [title, setTitle] = useState('');
+  const [todoUserId, setTodoUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setTodoUserId(0);
+    setHasTitleError(false);
+    setHasUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!todoUserId || !title.trim()) {
+      if (!title.trim()) {
+        setHasTitleError(true);
+      }
+
+      if (todoUserId === 0) {
+        setHasUserError(true);
+      }
```

Comment:
also, u case write this condition inside a setState `setHasUserError(todoUserId === 0);`